### DISCUSSION TITLE
Welcome to WhatTodo Discussions!

### DISCUSSION THREAD

**Discussion by burhanrashid52** [burhanrashid52/WhatTodo#32] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mohsenzahab** [comment]
Hello. What a perfect and clean code bro. Can you please explain what is happening when a new task created? Are you gonna read all tasks from DB? Because I can't see how you dealing with returned ids. I can't figure it out.

**Comment by burhanrashid52** [comment]
Thank you. I did not understand your question. Can you elaborate on what you are trying to achieve?

**Reply by mohsenzahab** [reply]
In my case, when I insert a new Task into DB, I will store the insert result (ID returned from Insert) in taks.id ( task.id=newId ). Because later id is used to updating or deleting the task. I wonder what are you doing with the returned ID because I don't see you save it in task.( I mean in task_db.dart file and the method updateTask )
**This is basically what I am doing:**
`Future<Task> insertTask(Task task) async {
    final db = await _DB.db;
    int id = await db.insert(TABLE_TASK, task.toMap());
    task.id = id;
     return task;
  }`
and then I will add this task to the list that will be shown to the user.
**and this is your code:**
`  Future updateTask(Tasks task, {List<int>? labelIDs}) async {
    var db = await _appDatabase.getDb();
    await db.transaction((Transaction txn) async {
      int id = await txn.rawInsert('INSERT OR REPLACE INTO '
          '${Tasks.tblTask}(${Tasks.dbId},${Tasks.dbTitle},${Tasks.dbProjectID},${Tasks.dbComment},${Tasks.dbDueDate},${Tasks.dbPriority},${Tasks.dbStatus})'
          ' VALUES(${task.id}, "${task.title}", ${task.projectId},"${task.comment}", ${task.dueDate},${task.priority.index},${task.tasksStatus!.index})');
      if (id > 0 && labelIDs != null && labelIDs.length > 0) {
        labelIDs.forEach((labelId) {
          txn.rawInsert('INSERT OR REPLACE INTO '
              '${TaskLabels.tblTaskLabel}(${TaskLabels.dbId},${TaskLabels.dbTaskId},${TaskLabels.dbLabelId})'
              ' VALUES(null, $id, $labelId)');
        });
      }
    });
  }`

**Reply by burhanrashid52** [reply]
Ok, I got it now. Your way of doing will also work. It is just that, this idea did not come to my mind while implementing it. I returned the task id for future use. But I did not use it. Your approach might be a good use case for that.
